### PR TITLE
Sk post login redirect

### PR DESCRIPTION
**Pull request by ShaneKilkelly**
Use session for post-login redirect

### CODE REVIEW

File: app/coffee/Features/Authentication/AuthenticationController.coffee [hunk 1]
```diff
@@ -157,21 +157,23 @@ module.exports = AuthenticationController =
 		return isValid
 
 	_redirectToLoginOrRegisterPage: (req, res)->
-		if req.query.zipUrl? or req.query.project_name?
+		if (req.query.zipUrl? or
+			  req.query.project_name? or
+			  req.path == '/user/subscription/new')
```
Comment (henryoswald) [inline app/coffee/Features/Authentication/AuthenticationController.coffee]: what is the reason to check if its a new subscription?

File: app/coffee/Features/Authentication/AuthenticationController.coffee [hunk 2]
```diff
@@ -188,3 +190,16 @@ module.exports = AuthenticationController =
 	_recordFailedLogin: (callback = (error) ->) ->
 		Metrics.inc "user.login.failed"
 		callback()
+
+	_setRedirectInSession: (req, value) ->
+		if !value?
+			value = if Object.keys(req.query).length > 0 then "#{req.path}?#{querystring.stringify(req.query)}" else req.path
```
Comment (henryoswald) [inline app/coffee/Features/Authentication/AuthenticationController.coffee]: assume querystring.stringify adds the ? on at the start

File: app/coffee/Features/Subscription/SubscriptionController.coffee [hunk 3]
```diff
@@ -14,10 +14,7 @@ module.exports = SubscriptionController =
 
 	plansPage: (req, res, next) ->
 		plans = SubscriptionViewModelBuilder.buildViewModel()
-		if AuthenticationController.isUserLoggedIn(req)
-			baseUrl = ""
-		else
-			baseUrl = "/register?redir="
```
Comment (henryoswald) [inline app/coffee/Features/Subscription/SubscriptionController.coffee]: is this /register not need any more. 

It looks like baseURL isn't used at all if that is the case.

File: app/coffee/Features/Authentication/AuthenticationController.coffee (hunk 1 above)
Comment (ShaneKilkelly) [inline app/coffee/Features/Authentication/AuthenticationController.coffee]: This ties in with the removal of baseUrl from the SubscriptionController. It used to direct specifically to the `register` page for the new subs, and the inclusion of `/user/subscription/new` does the same here.

File: app/coffee/Features/Authentication/AuthenticationController.coffee (hunk 2 above)
Comment (ShaneKilkelly) [inline app/coffee/Features/Authentication/AuthenticationController.coffee]: Will check.

File: app/coffee/Features/Subscription/SubscriptionController.coffee (hunk 3 above)
Comment (ShaneKilkelly) [inline app/coffee/Features/Subscription/SubscriptionController.coffee]: probably true, will check and remove completely.

File: app/coffee/Features/Authentication/AuthenticationController.coffee (hunk 1 above)
Comment (henryoswald) [inline app/coffee/Features/Authentication/AuthenticationController.coffee]: great

File: app/coffee/Features/Authentication/AuthenticationController.coffee (hunk 2 above)
Comment (ShaneKilkelly) [inline app/coffee/Features/Authentication/AuthenticationController.coffee]: Nope:

```
nesh> querystring.stringify({a: 1})
'a=1'
```

**Pull request merged**